0.2-Elliptic-Curve-Math.ipynb

```diff
@@ -0,0 +1,303 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import random\n",
+    "import util\n",
+    "from test_framework.key import SECP256K1_ORDER, SECP256K1_FIELD_SIZE, ECKey, ECPubKey"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 0.2 Elliptic Curve Math (Review)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Elliptic Curve math involves scalars and Points. \n",
+    "* A scalar is a positive integer which is smaller than secp256k1 the field size. \n",
+    "* A point `P(x,y)` lies on the secp256k1 curve.\n",
+    "\n",
+    "The following provides an overview of all operations of scalars and points over elliptic curves. In Bitcoin, key pair generation and signing is performed over the SECP256k1 curve.\n",
+    "\n",
+    "![test](images/ec_math0.jpg)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Classes / Methods for Elliptic Curve Math\n",
+    "\n",
+    "**`Integers`:** All Scalar operations over SECP251k1 can be performed with python integers and the modulo `%` operator. \n",
+    "\n",
+    "Scalar addition, subtraction, multiplication and division over SECP256k1 are modulo a large prime number SECP256K1_ORDER. \n",
+    "\n",
+    "\n",
+    "* All scalar operations are performed modulo n (`SECP256K1_ORDER`).\n",
+    "* Addition: `a + b % SECP256K1_ORDER`\n",
+    "* Subtraction: `-a = SECP256K1_ORDER - a`\n",
+    "* Multiplication: `a * b % SECP256K1_ORDER`\n",
+    "* Division (Fermat's little Theorem): `1/b = b ** (SECP256K1_ORDER-2) % SECP256K1_ORDER`\n",
+    "\n",
+    "**`ECKey`:** The Bitcoin Core library provides a private key class which can also perform certain scalar operations.\n",
+    "* Addition: `a + b`\n",
+    "* Subtraction: \n",
+    "    * `b.negate()`\n",
+    "    * `a + b` \n",
+    "* Multiplication: `a * b`\n",
+    "* Division: `a * 1/b` (See Fermat's little theorem) \n",
+    "\n",
+    "![test](images/ec_math1.jpg)"
```

Comment:
The first two columns contain `ECKey * ECKey`. I think they should say `ECKey + ECKey`